lettuce/cuda_native/_template.py

```diff
@@ -264,34 +296,8 @@ def invoke(simulation):
   {pipeline_buffer}
 
 #pragma unroll
-  for (index_t i = 0; i < q; ++i) {{
-
-#if {support_no_streaming_mask}
-    if (no_streaming_mask[dist_index[i]])
-      f_next[dist_index[i]] = f_reg[i];
-    else
-#endif
-
-    {{
-      index_t neighbor_x = index[0] + e[i][0];
-           if (neighbor_x <  0)            neighbor_x += dimension[0];
-      else if (neighbor_x >= dimension[0]) neighbor_x -= dimension[0];
-
-#if d > 1
-      index_t neighbor_y = index[1] + e[i][1];
-           if (neighbor_y <  0)            neighbor_y += dimension[1];
-      else if (neighbor_y >= dimension[1]) neighbor_y -= dimension[1];
-#endif
-
-#if d > 2
-      index_t neighbor_z = index[2] + e[i][2];
-           if (neighbor_z <  0)            neighbor_z += dimension[2];
-      else if (neighbor_z >= dimension[2]) neighbor_z -= dimension[2];
-#endif
-
-      f_next[dist_coord(i, neighbor_x, neighbor_y, neighbor_z)] = f_reg[i];
-    }}
-  }}
+  for (index_t i = 0; i < q; ++i)
+    write(i);
```

Comment:
key change: simplified template body by shifting the streaming logic into a macro definition